### PR TITLE
*: add a meta column in value indicate whether it should be ignore by ticdc

### PR DESCRIPTION
**Pull request by xiongjiwei**
Signed-off-by: xiongjiwei <xiongjiwei1996@outlook.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/38587

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [x] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -1033,6 +1033,10 @@ type SessionVars struct {
 
 	// MetricSchemaStep indicates the step when query metric schema.
 	MetricSchemaStep int64
+
+	// TiDBWriteByTiCDC indicates whether the following data is written by TiCDC.
+	TiDBWriteByTiCDC bool
```
Comment (tangenta) [inline sessionctx/variable/session.go:1038]:
```suggestion
	// WriteByTiCDC indicates whether the following data is written by TiCDC.
	WriteByTiCDC bool
```

Please keep consistent with other variables.

File: table/tables/tables.go [hunk 2]
```diff
@@ -540,6 +542,18 @@ func (t *TableCommon) rebuildIndices(ctx sessionctx.Context, txn kv.Transaction,
 	return nil
 }
 
+func appendMetaCol(sctx sessionctx.Context, colIDs []int64, row []types.Datum) ([]int64, []types.Datum) {
+	if shouldWriteMetaCol(sctx) {
+		colIDs = append(colIDs, model.ExtraMetaColID)
+		row = append(row, types.NewMetaDatum(sctx.GetSessionVars().TiDBWriteByTiCDC))
```
Comment (tangenta) [inline table/tables/tables.go:548]:
```suggestion
		row = append(row, types.NewMetaDatum(sctx.GetSessionVars().TiDBWriteByTiCDC))
```

`sctx.GetSessionVars().TiDBWriteByTiCDC` is always true.

Comment (xiongjiwei) [reply table/tables/tables.go]: done

**Review by tangenta — APPROVED** [review]

File: table/tables/tables.go [hunk 3]
```diff
@@ -427,6 +427,8 @@ func (t *TableCommon) UpdateRecord(ctx context.Context, sctx sessionctx.Context,
 		}
 	}
 
+	colIDs, row = appendMetaCol(sctx, colIDs, row)
```
Comment (bb7133) [inline table/tables/tables.go:430]: I see `appendMetaCol` is added to `UpdateRecord()` and `AddRecord()`, how about `RemoveRecord`?

**Pull request closed without merging**